pkg/resources/kafka/kafka.go

```diff
@@ -804,3 +798,25 @@ func GetBrokersWithPendingOrRunningCCTask(kafkaCluster *v1beta1.KafkaCluster) []
 func isDesiredStorageValueInvalid(desired, current *corev1.PersistentVolumeClaim) bool {
 	return desired.Spec.Resources.Requests.Storage().Value() < current.Spec.Resources.Requests.Storage().Value()
 }
+
+func (r *Reconciler) createExternalListenerStatuses() (map[string]v1beta1.ListenerStatus, error) {
+	extListenerStatuses := make(map[string]v1beta1.ListenerStatus, len(r.KafkaCluster.Spec.ListenersConfig.ExternalListeners))
+	for _, eListener := range r.KafkaCluster.Spec.ListenersConfig.ExternalListeners {
+		var host string
+		if eListener.HostnameOverride != "" {
+			host = eListener.HostnameOverride
+		} else if eListener.GetAccessMethod() == corev1.ServiceTypeLoadBalancer {
+			lbIP, err := getLoadBalancerIP(r.Client, r.KafkaCluster.GetNamespace(),
+				r.KafkaCluster.Spec.GetIngressController(), r.KafkaCluster.GetName(), eListener.Name)
+			if err != nil {
+				return nil, err
+			}
+			host = lbIP
+		}
+		extListenerStatuses[eListener.Name] = v1beta1.ListenerStatus{
+			Host: host,
+			Port: eListener.ContainerPort,
```

Comment:
`ContainerPort` should be used only with `NodePort` when specif external IP is provided.